src/nginx/grpc.cc

```diff
@@ -227,6 +227,8 @@ ngx_int_t GrpcBackendHandler(ngx_http_request_t *r) {
     }
   } else if (ctx && ctx->request_handler && IsGrpcWeb(r)) {
     ctx->grpc_backend = true;
+    ctx->request_handler->TryAddApiKeyHeaderFromQuery();
```

Comment:
Check grpc web format to see if it is possible that it can have query parameters?  If no, no need to call this.